src/Components/Web/src/Forms/InputLargeTextArea/InputLargeTextArea.cs

```diff
@@ -0,0 +1,149 @@
+            {
+                _jsCallbacksRelay = new InputLargeTextAreaJsCallbacksRelay(this);
+                await JSRuntime.InvokeVoidAsync(InputLargeTextAreaInterop.Init, _jsCallbacksRelay.DotNetReference, _inputLargeTextAreaElement);
+            }
+        }
+
+        /// <inheritdoc/>
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            builder.OpenElement(0, "textarea");
+            builder.AddMultipleAttributes(1, AdditionalAttributes);
+            builder.AddElementReferenceCapture(2, elementReference => _inputLargeTextAreaElement = elementReference);
+            builder.CloseElement();
+        }
+
+        Task IInputLargeTextAreaJsCallbacks.NotifyChange(int length)
+            => OnChange.InvokeAsync(new InputLargeTextAreaChangeEventArgs(this, length));
+
+        /// <summary>
+        /// Retrieves the textarea value asyncronously.
+        /// </summary>
+        /// <param name="maxLength">The maximum length of content to fetch from the textarea.</param>
+        /// <param name="cancellationToken">The <see cref="System.Threading.CancellationToken"/> used to relay cancellation of the request.</param>
+        /// <returns>A <see cref="System.IO.TextReader"/> which facilitates reading of the textarea value.</returns>
+        public async ValueTask<StreamReader> GetTextAsync(int maxLength = 32_000, CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                var streamRef = await JSRuntime.InvokeAsync<IJSStreamReference>(InputLargeTextAreaInterop.GetText, cancellationToken, _inputLargeTextAreaElement);
+                var stream = await streamRef.OpenReadStreamAsync(maxLength, cancellationToken);
+                var streamReader = new StreamReader(stream);
+                return streamReader;
+            }
+            catch (JSException jsException)
+            {
+                // Special casing support for empty textareas. Due to security considerations
+                // 0 length streams/textareas aren't permitted from JS->.NET Streaming Interop.
+                if (jsException.InnerException is ArgumentOutOfRangeException)
+                {
+                    return StreamReader.Null;
+                }
+
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Sets the textarea value asyncronously.
+        /// </summary>
+        /// <param name="streamWriter">A <see cref="System.IO.StreamWriter"/> used to set the value of the textarea.</param>
+        /// <param name="leaveTextAreaEnabled">Don't disable the textarea while settings the new textarea value from the stream.</param>
+        /// <param name="cancellationToken">The <see cref="System.Threading.CancellationToken"/> used to relay cancellation of the request.</param>
+        public async ValueTask SetTextAsync(StreamWriter streamWriter, bool leaveTextAreaEnabled = false, CancellationToken cancellationToken = default)
```

Comment:
Missing `<returns>`?

Reply:
Is there another example of a component we allow deriving that exposes non-virtual public API? Wanted to find out if we should make the GetAsync / SetAsync virtual since a derived type might want to change it.

Reply:
> Wanted to find out if we should make the GetAsync / SetAsync virtual since a derived type might want to change it.

I think that makes sense. Updated.